### DISCUSSION TITLE
Why Prometheus metrics don't have no values?

### DISCUSSION THREAD

**Discussion by AdrianXIA** [keycloak/keycloak#24531] [Q&A]
Hello!

I tried to enable  Prometheus metrics for Keycloak **(v22.0.5)**. By accessing  the endpoint **auth/realm/<realm_name>/metrics**, metrics have no value:
![image](https://github.com/keycloak/keycloak/assets/7112324/dc337c03-788f-4e6c-a58b-1ad5ca03fe98).

I configured:
-  variable KC_METRICS_ENABLED=true;
- the jar is present inside container 
- ![image](https://github.com/keycloak/keycloak/assets/7112324/a9427802-82a9-4c58-a584-221e68db5886)
 the listener: metrics-listener is configured
![image](https://github.com/keycloak/keycloak/assets/7112324/a3338afb-dc12-44e3-93f9-a0b018765ba6)

-  no errors in container.

Which can be the reason?

Thank you!

**Comment by sschu** [comment]
Are you using this SPI https://github.com/aerogear/keycloak-metrics-spi? You might want to ask your question there as this was created externally. There were some changes with metrics inside Keycloak so maybe this extension does not work anymore.

**Reply by AdrianXIA** [reply]
I will put there the question also. Thank you!

**Reply by Srikanth-77772** [reply]
Can we know how to get username metrics who has logged in application 

Thanks in adavance
